pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -360,16 +369,45 @@ func TryUpdatingCache(ctx *types.SystemContext) (*V2RegistriesConf, error) {
 	defer configMutex.Unlock()
 
 	// load the config
-	config, err := loadRegistryConf(configPath)
-	if err != nil {
+	config := &tomlConfig{}
+	if err := config.loadConfig(configPath); err != nil {
 		// Return an empty []Registry if we use the default config,
 		// which implies that the config path of the SystemContext
 		// isn't set.  Note: if ctx.SystemRegistriesConfPath points to
 		// the default config, we will still return an error.
 		if os.IsNotExist(err) && (ctx == nil || ctx.SystemRegistriesConfPath == "") {
-			return &V2RegistriesConf{Registries: []Registry{}}, nil
+			config = &tomlConfig{}
+			config.V2RegistriesConf = V2RegistriesConf{Registries: []Registry{}}
+		} else {
+			return nil, err
 		}
-		return nil, err
+	}
+
+	// Load the configs from the conf directory path.
+	confDir := systemRegistriesConfDirPath
+	if ctx != nil && ctx.SystemRegistriesConfDirPath != "" {
+		confDir = ctx.SystemRegistriesConfDirPath
+	}
+	err := filepath.Walk(confDir,
+		// WalkFunc to read additional
+		func(path string, info os.FileInfo, err error) error {
+			switch {
+			case info == nil:
+				// registries.conf.d doesn't exist
+				return nil
+			case info.IsDir():
+				// ignore directories
+				return nil
+			default:
```

Comment:
Consider restricting this to only `*.conf` files, so that random editor backups and WIP temporary files are not read.

Reply:
Done.